### PR TITLE
Generate datamapper real time data using input file

### PR DESCRIPTION
**Pull request by dinuish94**

### CODE REVIEW

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/part/DataMapperDiagramEditor.java [hunk 1]
```diff
@@ -600,6 +609,45 @@ private void updateSchema() {
 		reloadDataMapperTestWindow(getInputSchemaType(), getOutputSchemaType());
 	}
 
+    /**
+     * Compares the schema before updating with the new schema to check whether there are any changes
+     * 
+     * @param modifiedSchema - changed schema
+     * @throws ParseException
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/part/DataMapperDiagramEditor.java]: Update java doc comments

Comment (dinuish94) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/part/DataMapperDiagramEditor.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java [hunk 2]
```diff
@@ -1012,6 +1035,24 @@ private void saveInputOutputSchema(String schemaFilePath) {
...
+     */
+    private void copyInputFile(String originalFilePath, String destinationFilePath) {
+        File originalFile = new File(originalFilePath);
+        File copiedFile = new File(destinationFilePath);
+        try {
+            FileUtils.copyFile(originalFile, copiedFile);
+            assert (copiedFile).exists();
+            assert (Files.readAllLines(originalFile.toPath()).equals(Files.readAllLines(copiedFile.toPath())));
+        } catch (IOException e) {
+            log.error("Failed to create a copy of the original input file!");
```
[the remark this answers is not in the document]
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/part/DataMapperDiagramEditor.java [hunk 3]
```diff
@@ -600,6 +609,45 @@ private void updateSchema() {
...
+            JSONObject prevSchemaJson = (JSONObject) new JSONParser().parse(prevSchema);
+            JSONObject modifiedSchemaJson = (JSONObject) new JSONParser().parse(modifiedSchema);
+            modifiedSchemaJson = (JSONObject) new JSONParser().parse(modifiedSchema);
+
+            // Compare whether the schema has any changes
+            if (prevSchemaJson != null && modifiedSchema != null && !prevSchemaJson.equals(modifiedSchemaJson)) {
+                return true;
+            }
+        } catch (ParseException | IOException e) {
+            log.error("Failed to compare schema!");
```
[the remark this answers is not in the document]
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/part/DataMapperDiagramEditor.java]: Fixed in #1063

**Pull request merged**